lib/Serialization/Serialization.cpp

```diff
@@ -579,10 +579,11 @@ void Serializer::writeHeader(const SerializationOptions &options) {
     ModuleName.emit(ScratchRecord, M->getName().str());
 
     SmallString<32> versionStringBuf;
+    auto version = M->getASTContext().LangOpts.EffectiveLanguageVersion;
     llvm::raw_svector_ostream versionString(versionStringBuf);
-    versionString << Version::getCurrentLanguageVersion();
+    versionString << version;
```

Comment:
I think it's "I can't load this module because it was built with the Swift %0 compiler", not "I can't load this module because it only supports Swift %0 (as a language)". We *are* planning to support "Swift 3 modules" in Swift 4, but only if they were built with the Swift 4 compiler.